storyCite.lua

```diff
@@ -0,0 +1,248 @@
+		local outputText = ""
+		if data["adapted from"] ~= "" then
+			outputText = outputText .. "Adapted from ''" .. data["adapted from"] .. "''"
+			if data["writer"] ~= "" then
+				outputText = outputText .. ", written by " .. data["writer"]
+				if data["publisher"] ~= "" then
+					outputText = outputText .. " and released by " .. data["publisher"]
+				else
+					outputText = outputText .. "and released "
+				end
+			elseif data["publisher"] ~= "" then
+				outputText = outputText .. " and released by " .. data["publisher"]
+			else
+				outputText = outputText .. "and released "
+			end
+		elseif data["writer"] ~= "" then
+			outputText = outputText .. "Written by " .. data["writer"]
+			if data["publisher"] ~= "" then
+				outputText = outputText .. " and released by " .. data["publisher"]
+			else
+				outputText = outputText .. "and released "
+			end
+		elseif data["publisher"] ~= "" then
+			outputText = outputText .. " Released by " .. data["publisher"]
+		else
+			outputText = outputText .. "Released "
+		end
+		if data["anthology"] ~= "" then
+				outputText = outputText .. " as part of ''" .. data["anthology"] .. "''"
+				if data["series"] ~= "" then
+					if (args["season number"] ~= "") or (args["season number"] ~= nil) then
+						outputText = outputText .. " from " .. data["series"]
+					else
+						outputText = outputText .. " from the series " .. data["series"]
+					end
+				end
+		elseif data["series"] ~= "" then
+			if (args["season number"] ~= "") or (args["season number"] ~= nil) then
+				outputText = outputText .. " as part of " .. data["series"]
+			else
+				outputText = outputText .. " as part of the series " .. data["series"]
+			end
+		end
+		if data["release year"] ~= "" then
+			outputText = outputText .. " " .. data["release year"]
+		end
+		if data["issues"] ~= "" then
+			outputText = outputText .. " throught " .. data["issues"]
+		end
+		--fallback in case of an error
+		if outputText == "" or outputText == nil then
+		 	outputText = "No data."
+		else
+			outputText = outputText .. "."
+		end
+		
+		--outputText = dump(data) --for testing
+		local name = frame.args[1]
+		local bin --used to store the output of the functions below
+		bin = util_vars.setVar(PREFIX .. name, outputText)
+		bin = cache.delete(PREFIX .. name)
+		bin = cache.set(PREFIX .. name, outputText)
+		bin = mw.smw.set("Story info", outputText) --documentation for this is down so I guessed the syntax. It doesn't seem to work.
+	end
+end
+
+function h.getInfo(story)
+	local info 
+	local bin --used to store the output of storing functions
+	if util_vars.getVar(PREFIX .. story) then
+		info = util_vars.getVar(PREFIX .. story)
+	elseif cache.get(PREFIX .. story) then
+		info = cache.get(PREFIX .. story)
+		bin = util_vars.setVar(PREFIX .. story, info)
+	elseif mw.smw.ask( story .. "\n?Story info") then
+		info = mw.smw.ask( story .. "\n?Story info")
+		bin = util_vars.setVar(PREFIX .. story, info)
+		bin = cache.delete(PREFIX .. name)
```

Comment:
you don't have to delete anything; we only got here if there was for certain no cache entry in the first place